bootstrap/kubeadm/config/default/kustomization.yaml

```diff
@@ -1,57 +1,126 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
 # Adds namespace to all resources.
 namespace: capi-kubeadm-bootstrap-system
 
 namePrefix: capi-kubeadm-bootstrap-
 
-commonLabels:
-  cluster.x-k8s.io/provider: "bootstrap-kubeadm"
+labels:
+- includeSelectors: true
+  pairs:
+    cluster.x-k8s.io/provider: bootstrap-kubeadm
 
 resources:
 - namespace.yaml
-
-bases:
 - ../crd
 - ../rbac
 - ../manager
 - ../webhook
 - ../certmanager
 
-patchesStrategicMerge:
-  # Provide customizable hook for make targets.
-  - manager_image_patch.yaml
-  - manager_pull_policy.yaml
-  # Enable webhook.
-  - manager_webhook_patch.yaml
-  # Inject certificate in the webhook definition.
-  - webhookcainjection_patch.yaml
-
-vars:
-  - name: CERTIFICATE_NAMESPACE # namespace of the certificate CR
-    objref:
-      kind: Certificate
-      group: cert-manager.io
-      version: v1
-      name: serving-cert # this name should match the one in certificate.yaml
-    fieldref:
-      fieldpath: metadata.namespace
-  - name: CERTIFICATE_NAME
-    objref:
-      kind: Certificate
-      group: cert-manager.io
-      version: v1
-      name: serving-cert # this name should match the one in certificate.yaml
-  - name: SERVICE_NAMESPACE # namespace of the service
-    objref:
-      kind: Service
-      version: v1
-      name: webhook-service
-    fieldref:
-      fieldpath: metadata.namespace
-  - name: SERVICE_NAME
-    objref:
-      kind: Service
-      version: v1
-      name: webhook-service
+patches:
+# Provide customizable hook for make targets.
+- path: manager_image_patch.yaml
+- path: manager_pull_policy.yaml
+# Enable webhook.
+- path: manager_webhook_patch.yaml
+# Inject certificate in the webhook definition.
+- path: webhookcainjection_patch.yaml
 
-configurations:
-  - kustomizeconfig.yaml
+replacements:
+- source: # Add cert-manager annotation to ValidatingWebhookConfiguration, MutatingWebhookConfiguration and CRDs
+    kind: Certificate
+    group: cert-manager.io
+    version: v1
+    name: serving-cert # this name should match the one in certificate.yaml
+    fieldPath: .metadata.namespace # namespace of the certificate CR
+  targets:
+    - select:
+        kind: ValidatingWebhookConfiguration
+      fieldPaths:
+        - .metadata.annotations.[cert-manager.io/inject-ca-from]
+      options:
+        delimiter: '/'
+        index: 0
+        create: true
```

Comment:
This way of selecting all resources of the relevant kind and then tell it to `create: true` can cause the annotation to be injected in resources where it is not needed. I think it would be better to specify the exact names that we target actually to avoid this.

To check if we inject something new with this patch I did the following:

```
git checkout <patch>
kustomize build config/default > new.yaml
git checkout main
kustomize build config/default > old.yaml

diff old.yaml new.yaml
```

The result looks like this (AMAZING work to preserve everything!), just two secrets got a slightly different name and the annotation was injected in 3 places where it didn't use to be.

```diff
❯ diff old.yaml new.yaml 
4581a4582
>     cert-manager.io/inject-ca-from: capi-system/capi-serving-cert
4851a4853
>     cert-manager.io/inject-ca-from: capi-system/capi-serving-cert
5127a5130
>     cert-manager.io/inject-ca-from: capi-system/capi-serving-cert
12588c12591
<           secretName: capi-webhook-service-cert
---
>           secretName: webhook-service-cert
12604c12607
<   secretName: capi-webhook-service-cert
---
>   secretName: webhook-service-cert
```

Could you try to make the diff completely go away? :pray:

Reply:
I added a `reject` field like this to the yaml and now the extra injection are not added. 
```
    select:
      kind: CustomResourceDefinition
    reject:
    - name: ipaddressclaims.ipam.cluster.x-k8s.io
    - name: ipaddresses.ipam.cluster.x-k8s.io
    - name: extensionconfigs.runtime.cluster.x-k8s.io
```